### PR TITLE
User Management

### PR DESCRIPTION
**Pull request by RandSohail**
#20

### CODE REVIEW

**Review by AymanSami98 — APPROVED** [review]

File: src/controllers/users/addUser.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      length: 15, numbers: true, strict: true, lowercase: true, uppercase: true,
+    });
+    const hashedPassword = await hash(password, 10);
+
+    const user = await addUser({
+      name,
+      email: email.toLowerCase(),
+      userRoleId: roleId,
+      password: hashedPassword,
+      createdBy: req.user?.id || MASTER_ADMIN,
```
Comment (Jood80) [inline src/controllers/users/addUser.ts:33]: OR, using nullish coalescing operator:
 `req.user.is ?? MASTER_ADMIN`
 If the left hand side was falsy value the right hand side will be selected.
 What are falsy values? see [this](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) and for more fun [check that](https://www.freecodecamp.org/news/falsy-values-in-javascript/)

Comment (Jood80) [inline src/controllers/users/addUser.ts:33]: What's the point of the `?` with `||` here anyway?

File: src/controllers/users/editProfile.ts [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
-    if (!currentUser) {
-      throw new CustomError(messages.authResponse.notExist, 404);
-    }
+    if (!currentUser) throw new CustomError(notExist, HttpStatus.NOT_FOUND);
 
     if (image) {
       const { Location } = await upload(image, id);
       userUpdatedFields.image = Location;
     }
+    userUpdatedFields.updatedBy = req.user?.id;
```
Comment (Jood80) [inline src/controllers/users/editProfile.ts:22]: Add a fallback id in case this dude was undefined

**Review by Jood80 — CHANGES_REQUESTED** [review]

**Comment by Jood80** [comment]
You have forgotten conflicts waiting to be solved to :)

**Review by Ibrahim-Jarada — APPROVED** [review]

**Review by zaher-aa — APPROVED** [review]

File: src/services/user/userStatus.ts [hunk 3]
```diff
@@ -11,6 +11,9 @@ const getUsersStatus: GetUsersStatus = (statusId: number) => User.findAll({
   where: {
     [Op.and]: [
       { userStatusId: statusId },
+<<<<<<< HEAD
```
Comment (Jood80) [inline src/services/user/userStatus.ts:14]: same here

File: src/helpers/validation/index.ts [hunk 4]
```diff
@@ -5,4 +5,9 @@ export { default as emailSchema } from './emailSchema';
 export { default as financialInformationSchema } from './financialInfoSchema';
 export { default as editProfileSchema } from './editProfileSchema';
 export { default as passwordSchema } from './passwordSchema';
+<<<<<<< HEAD
+export { default as createUserSchema } from './createUserSchema';
+export { default as getUsersAndContentsSchema } from './getUsersAndContentsSchema';
+=======
 export { default as getPaginatedDataSchema } from './getPaginatedDataSchema';
+>>>>>>> 255f3e04629fbbbbf0e8f54d6f1fb87e6153d7c6
```
Comment (Jood80) [inline src/helpers/validation/index.ts:13]: Are these lines okay to have?

**Review by Jood80 — APPROVED** [review]

**Pull request merged**